tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,214 @@
+import os
+import stat
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Optional, TextIO, Union
+
+import pytest
+
+from ert._c_wrappers.enkf import ErtConfig
+from ert.parsing import ConfigValidationError
+
+test_config_file_base = "test"
+test_config_filename = f"{test_config_file_base}.ert"
+
+
+@dataclass
+class FileDetail:
+    contents: str
+    is_executable: bool = False
+
+
+def assert_error_from_config_with(
```

Comment:
I don't understand the function name. The function seems to do too much. I think it would be better to split this up into more functions, probably have the file writing in the test rather than in a function.

Reply:
Addressed by renaming the function and splitting file writing into own function. I think it is OK to have the file writing outside the test declaration itself as it makes it only declare the contents of the config and the expected error. As for the order of execution / logic packed into the function, each step depends on the step before as it (1) finds all errors with the given filename, then (2) finds errors matching the expected location by line & start/end columns, then (3) finds errors among those matching the given string (if given). Since there will be quite a bit of these tests, several involving file writing etc I'd suggest keeping the file writing inside this function as to not end up copypasting it into several places.